### PR TITLE
feat: add support for specifying LoadBalancer class name

### PR DESCRIPTION
**Pull request by jamezrin**
This PR adds support for configuring the loadBalancerClass via the chart values.

### CODE REVIEW

**Comment by jamezrin** [comment]
I have read the CLA Document and I hereby sign the CLA

File: helm/coder/values.yaml [hunk 1]
```diff
@@ -281,6 +281,9 @@ coder:
     # your cloud and specify it here in production to avoid accidental IP
     # address changes.
     loadBalancerIP: ""
+    # coder.service.className -- The class name of the LoadBalancer. See:
+    # https://kubernetes.io/docs/concepts/services-networking/service/#load-balancer-class
+    className: ""
```
Comment (deansheather) [inline helm/coder/values.yaml:286]: Can this be changed to `loadBalancerClass` to match the field in the k8s yaml?

**Review by johnstcn — CHANGES_REQUESTED** [review]
Thanks for your contribution!

Before merging, please add some new test cases to cover this new field:

- Add a corresponding values YAML and golden file in `helm/coder/tests/testdata/`. You can copy an existing golden file and modify it to suit.
- Add the corresponding test case in `helm/coder/tests/chart_test.go`
- Modify the golden file and/or changes until the output is as expected.

I would also suggest modifying the service template such that if new field is not specified, no changes should be made to existing test cases.

I also agree with @deansheather 's suggestion regarding the naming of the new field.

Thanks!

**Comment by jamezrin** [comment]
Thanks for reviewing, I made the changes. The tests are passing locally, but I don't think the CI is running them, is that normal?

File: helm/coder/values.yaml (hunk 1 above)
Comment (jamezrin) [reply helm/coder/values.yaml]: Done

**Comment by johnstcn** [comment]
> Thanks for reviewing, I made the changes. The tests are passing locally, but I don't think the CI is running them, is that normal?

It is! I just pushed the button to make them run.

**Review by deansheather — APPROVED** [review]

**Review by johnstcn — APPROVED** [review]

**Comment by johnstcn** [comment]
Merged, thanks @jamezrin !

**Pull request merged**